model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -63,8 +64,19 @@ object TagInliner extends Logging {
         )
 
       case ForTag(item, iterable) =>
-        valueToModel(iterable).map { case (v, p) =>
-          Some(ForModel(item, v)) -> p
+        for {
+          vp <- valueToModel(iterable)
+          (v, p) = vp
+          exps <- Exports[S].exports
+          n <- Mangler[S].findAndForbidName(item)
+          elementType = iterable.`type` match {
+            case b: BoxType => b.element
+            // TODO: it is unexpected, should we handle this?
+            case _ => iterable.`type`
```

Comment:
Maybe add error log?
I suppose we might add the box type to `ForTag`, as it was resolved there.